x-pack/plugins/observability_solution/synthetics/server/synthetics_service/service_api_client.ts

```diff
@@ -215,21 +215,43 @@ export class ServiceAPIClient {
 
     const monitorsByLocation = this.processServiceData(serviceData);
 
-    monitorsByLocation.forEach(({ location: { url, id }, monitors, data }) => {
-      const promise = this.callServiceEndpoint(data, method, url, endpoint);
-      promises.push(
-        rxjsFrom(promise).pipe(
+    monitorsByLocation.forEach(({ location: { url, id }, data }) => {
+      const sendRequest = (payload: ServicePayload): Observable<any> => {
+        const promise = this.callServiceEndpoint(payload, method, url, endpoint);
+        return rxjsFrom(promise).pipe(
           tap((result) => {
-            this.logSuccessMessage(url, method, monitors.length, result);
+            this.logSuccessMessage(url, method, payload.monitors.length, result);
           }),
           catchError((err: AxiosError<{ reason: string; status: number }>) => {
+            if (err.response?.status === 413 && payload.monitors.length > 1) {
```

Comment:
We should make this logic recursive instead of a single retry.

Reply:
it's already recursive